qiskit_experiments/characterization/t2ramsey.py

```diff
@@ -99,3 +102,203 @@ def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+    t2ramsey.T2Ramsey` by the sensitivity to the sign of frequency error.
+    This experiment consists of following two circuits:
+
+    .. parsed-literal::
+
+        (Ramsey X) Second pulse is SX
+
+                   ┌────┐┌─────────────┐┌────┐ ░ ┌─┐
+              q_0: ┤ √X ├┤ Delay(τ[s]) ├┤ √X ├─░─┤M├
+                   └────┘└─────────────┘└────┘ ░ └╥┘
+        measure: 1/═══════════════════════════════╩═
+                                                  0
+
+        (Ramsey Y) Second pulse is SY
+
+                   ┌────┐┌─────────────┐┌──────────┐┌────┐ ░ ┌─┐
+              q_0: ┤ √X ├┤ Delay(τ[s]) ├┤ Rz(-π/2) ├┤ √X ├─░─┤M├
+                   └────┘└─────────────┘└──────────┘└────┘ ░ └╥┘
+        measure: 1/═══════════════════════════════════════════╩═
+                                                              0
+
+    The first (second) circuit measures :math:`\sigma_Y` (:math:`\sigma_X`) expectation value,
+    so this experiment draws the dynamics of the Bloch vector as if drawing a Lissajous figure.
+
+    Given the control electronics tracks the frame of qubit at the reference frequency
+    which is slightly differ from the true qubit frequency by :math:`\Delta\omega`,
+    and the IQ mixer skew can be ignored, we can describe the dynamics of
+    two circuits as follows.
+
+    The Hamiltonian during the ``Delay`` instruction can be written as
+    :math:`H^R = - \frac{1}{2} \Delta\omega` in the rotating frame,
+    and the propagator will be :math:`U(\tau) = \exp(-iH^R\tau)` where :math:`\tau` is the
+    duration of the delay. By scanning this duration, we can get
+
+    .. math::
+
+        {\cal E}_x(\tau)
+            = {\rm Re} {\rm Tr}\left( \sigma_Y U \rho U^\dagger \right)
+            &= - \cos(\Delta\omega\tau) = \sin(\Delta\omega\tau - \frac{\pi}{2}), \\
+        {\cal E}_y(\tau)
+            = {\rm Re} {\rm Tr}\left( \sigma_X U \rho U^\dagger \right)
+            &= \sin(\Delta\omega\tau),
+
+    where :math:`\rho = | L \rangle` prepared by the :math:`\sqrt{\rm X}` gate in the beginning.
+
+    Note that phase difference of these two outcomes :math:`{\cal E}_x, {\cal E}_y` depends on
+    the frequency offset :math:`\Delta\omega`, which is also sensitive
+    to the sign of it, in contrast to the standard Ramsey experiment which
+    usually consists only of the first circuit,
+    i.e. :math:`\cos(-\Delta\omega\tau) = \cos(\Delta\omega\tau)`.
+    """
+
+    __analysis_class__ = RamseyXYAnalysis
+
+    def __init__(
+            self,
+            qubit: int,
+            delays: Union[Iterable[float], np.ndarray],
+            unit: str = "s",
+            offset_frequency: float = 0.,
+    ):
+        """Initialize new experiment.
+
+        Args:
+            qubit: The qubit under test.
+            delays: Delay times of the experiments.
+            unit: Time unit of `delays`. One of 's', 'ms', 'us', 'ns', 'ps', 'dt'.
+            offset_frequency: Frequency offset for this experiment.
```

Comment:
We have some inconsistency in the way we are setting such options. E.g. in Drag and Rabi they are not given in the init but in spec they are given in the init. We should chose one way and align the other experiments accordingly. I don't have a strong preference but having them in the init for Drag and Rabi could save the user a line of code. Thoughts?

Reply:
I think having these settings in the constructor is good for single liner execution that is one of concepts of the experiment. On the other hand, I don't prefer saving these settings as a instance variable, i.e. `self.` since these settings are also stored in the experiment options. This could induce some edge cases. This is why I update the experiment options in the constructor and discard these values.